pkg/metrics/metrics.go

```diff
@@ -110,6 +110,11 @@ func Validate(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not get metrics Service monitor")
 	}
 
+	// Remove any existing stale resources from previous operator version.
+	if err := removeInvalidResources(ctx, cfg, namespace); err != nil {
+		return errors.Wrap(err, "could not remove invalid resources")
+	}
```

Comment:
This seems like it would be better suited at the start of the function instead of the end.
Even if it doesnt matter functionally now, it gives us a clean slate to work with